interpreter/src/main/scala/pl/touk/nussknacker/engine/Interpreter.scala

```diff
@@ -220,29 +210,38 @@ object Interpreter {
     new Interpreter(listeners, expressionEvaluator, runMode)
   }
 
+  object InterpreterShape {
+    def transform[T, E](f: Future[T])(mapError: Throwable => E)(implicit ec: ExecutionContext): Future[Either[T, E]] = f.transformWith {
+      case Failure(exception) => Future.successful(Right(mapError(exception)))
+      case Success(value) => Future.successful(Left(value))
+    }
+  }
+
   //Interpreter can be invoked with various effects, we require MonadError capabilities and ability to convert service invocation results
   trait InterpreterShape[F[_]] {
 
-    def monadError: MonadError[F, Throwable]
+    def monad: Monad[F]
 
-    def fromFuture[T](implicit ec: ExecutionContext): Future[T] => F[T]
+    def fromFuture[T, E](handleError: Throwable => E)(implicit ec: ExecutionContext): Future[T] => F[Either[T, E]]
```

Comment:
Can we just make 
`fromFuture[T](implicit ec:ExecutionContext): F[Either[T, Throwable]]` and do `transform` as `map` on F?